mita/openstack.py

```diff
@@ -211,28 +237,20 @@ def _get_network(self, network: Optional[str] = None) -> OpenStackNetwork:
             else [network]
         )
 
-        _available_networks = self.driver_v2.ex_list_networks()
-
         for net in _networks:
+            # Treating an exception as a soft error as it is possible to find another
+            # suitable network from the list.
             try:
-                os_net = [n for n in _available_networks if n.name == net][0]
+                os_net = self._get_network_by_name(name=net)
 
-                if not self._has_free_ip_address(os_net.id):
-                    logger.debug("%s does not meet the requirements", net)
+                if not self._has_free_ip_address(net=os_net):
+                    logger.debug("%s has no free ip addresses.", os_net.name)
```

Comment:
To make the logs easier to read, let's take this `logger.debug()` statement out and only log the network names if we hit `ResourceNotFound` at the end.

Reply:
Done